hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -729,6 +733,12 @@ object HoodieSparkSqlWriter {
         mergedParams(key) = value
       }
     }
+
+    // use preCombineField to fill in PAYLOAD_ORDERING_FIELD_PROP_KEY and PAYLOAD_EVENT_TIME_FIELD_PROP_KEY
+    if (mergedParams.contains(PRECOMBINE_FIELD.key())) {
+      mergedParams.put(HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY, mergedParams(PRECOMBINE_FIELD.key()))
+      mergedParams.put(HoodiePayloadProps.PAYLOAD_EVENT_TIME_FIELD_PROP_KEY, mergedParams(PRECOMBINE_FIELD.key()))
+    }
```

Comment:
the expected scenario should be: when `preCombineField` is set, `PAYLOAD_ORDERING_FIELD_PROP_KEY` is set automatically, and PAYLOAD_EVENT_TIME_FIELD_PROP_KEY is never required. It'll be good to fix these along the way.

Reply:
but I see it didn't do that and I don't know how can run correctly before.

btw, why `PAYLOAD_EVENT_TIME_FIELD_PROP_KEY ` is never required. After all, `DefaultHoodiePayload` will use this for now.